projects/packages/forms/src/dashboard/inbox/response.js

```diff
@@ -21,48 +28,42 @@ const InboxResponse = () => {
 				/>
 			</div>
 
-			<h3 className={ titleClasses }>{ __( 'Bill Suitor', 'jetpack-forms' ) }</h3>
-			<p className="jp-forms__inbox-response-subtitle">
-				{ __( 'bill@la84.com', 'jetpack-forms' ) }
-			</p>
+			<h3 className={ titleClasses }>{ getDisplayName( response ) }</h3>
+			<p className="jp-forms__inbox-response-subtitle">{ response.author_email }</p>
 
 			<div className="jp-forms__inbox-response-meta">
 				<div className="jp-forms__inbox-response-meta-label">
-					{ __( 'Date:', 'jetpack-forms' ) }
-				</div>
-				<div className="jp-forms__inbox-response-meta-value">
-					{ sprintf(
-						/* Translators: %1$s is the date, %2$s is the time. */
-						__( '%1$s at %2$s', 'jetpack-forms' ),
-						dateI18n( getDateSettings().formats.date, new Date().getTime() ),
-						dateI18n( getDateSettings().formats.time, new Date().getTime() )
-					) }
+					{ __( 'Added on', 'jetpack-forms' ) }&nbsp;
+					<span className="jp-forms__inbox-response-meta-value">
+						{ sprintf(
+							/* Translators: %1$s is the date, %2$s is the time. */
+							__( '%1$s at %2$s', 'jetpack-forms' ),
+							dateI18n( getDateSettings().formats.date, response.date ),
+							dateI18n( getDateSettings().formats.time, response.date )
+						) }
+					</span>
```

Comment:
Not a blocker, but I would recommend making this a single translatable string. In some languages the date comes before "Added on".

Same for the other elements below.